simplecpp.cpp

```diff
@@ -1839,8 +1839,37 @@ namespace simplecpp {
 }
 
 #ifdef SIMPLECPP_WINDOWS
+
+class RealFileNameMap
+{
+public:
+    bool getRealPathFromCache(std::string const& path, std::string& returnPath)
+    {
+        std::unique_lock<std::mutex> lock(mMutex);
+        
+        auto it = mFileMap.find(path);
+        if (it != mFileMap.end())
+        {
+            returnPath = it->second;
+            return true;
+        }
+        return false;
+    }
+    
+    void addToCache(std::string const& path, std::string const& actualPath)
+    {
+        std::unique_lock<std::mutex> lock(mMutex);
+        mFileMap[path] = actualPath;
+    }
+    
+    std::map<std::string, std::string> mFileMap;
+    std::mutex mMutex;
+};
+
 static bool realFileName(const std::string &f, std::string *result)
 {
+    static RealFileNameMap realFileNameMap;
```

Comment:
This is not thread safe, as far as I know. I am pretty sure that the `realFileNameMap` will be initialized when this function is executed the first time (without locking mechanism). It should be safer if the variable is put in global scope.